js_modules/dagit/packages/core/src/app/AppTopNav.tsx

```diff
@@ -11,89 +11,126 @@ import {LayoutContext} from './LayoutProvider';
+          <ShortcutHandler
+            key="overview"
+            onShortcut={() => history.push('/overview')}
+            shortcutLabel="⌥1"
+            shortcutFilter={(e) => e.altKey && e.code === 'Digit1'}
           >
-            <Box flex={{direction: 'row', alignItems: 'center', gap: 6}}>
-              Deployment
-              <DeploymentStatusIcon />
-            </Box>
-          </TopNavLink>
-        </ShortcutHandler>
-      </Box>
-    );
+            <TopNavLink to="/overview" data-cy="AppTopNav_StatusLink">
+              Overview
+            </TopNavLink>
+          </ShortcutHandler>
+        ),
+      },
+      {
+        title: 'runs',
+        element: (
+          <ShortcutHandler
+            onShortcut={() => history.push('/instance/runs')}
+            shortcutLabel="⌥2"
+            shortcutFilter={(e) => e.altKey && e.code === 'Digit2'}
+          >
+            <TopNavLink to="/instance/runs" data-cy="AppTopNav_RunsLink">
+              Runs
+            </TopNavLink>
+          </ShortcutHandler>
+        ),
+      },
+      {
+        title: 'assets',
+        element: (
+          <ShortcutHandler
+            onShortcut={() => history.push('/instance/assets')}
+            shortcutLabel="⌥3"
+            shortcutFilter={(e) => e.altKey && e.code === 'Digit3'}
+          >
+            <TopNavLink to="/instance/assets" data-cy="AppTopNav_AssetsLink" exact={false}>
+              Assets
+            </TopNavLink>
+          </ShortcutHandler>
+        ),
+      },
+      {
+        title: 'workspace',
+        element: (
+          <ShortcutHandler
+            key="workspace"
+            onShortcut={() => history.push('/workspace')}
+            shortcutLabel="⌥4"
+            shortcutFilter={(e) => e.altKey && e.code === 'Digit4'}
+          >
+            <TopNavLink to="/workspace" data-cy="AppTopNav_WorkspaceLink">
+              Workspace
+            </TopNavLink>
+          </ShortcutHandler>
+        ),
+      },
+      {
+        title: 'deployment',
+        element: (
+          <ShortcutHandler
+            key="deployment"
+            onShortcut={() => history.push('/instance')}
+            shortcutLabel="⌥5"
+            shortcutFilter={(e) => e.altKey && e.code === 'Digit5'}
+          >
+            <TopNavLink
+              to="/instance"
+              data-cy="AppTopNav_StatusLink"
+              isActive={(_, location) => {
+                const {pathname} = location;
+                return (
+                  pathname.startsWith('/instance') &&
+                  !pathname.startsWith('/instance/runs') &&
+                  !pathname.startsWith('/instance/assets')
+                );
+              }}
+            >
+              <Box flex={{direction: 'row', alignItems: 'center', gap: 6}}>
+                Deployment
+                <DeploymentStatusIcon />
+              </Box>
+            </TopNavLink>
+          </ShortcutHandler>
+        ),
+      },
+    ];
```

Comment:
Since the items are defined statically, another possibility might be to export the nav link objects and optionally specify them at the callsite as a prop, so that we don't need to define the list of items within `AppTopNav` and pass them upward to be manipulated.

```
// from AppTopNav (or adjacent)
export const items = {
  overview: {
    path: '/overview',
    isActive: (_, location) => ...,
    label: ...,
  },
  // etc.
};

// callsite with custom items
const itemsToUse = [items.overview, items.runs, items.assets, ...];
<AppTopNav items={itemsToUse} ... />

// callsite with default items
<AppTopNav ... />

// cloud callsite
const mySetupItem = {...};
const itemsToUse = [mySetupItem, items.runs, items.assets, ...];
<AppTopNav items={itemsToUse} ... />
```

This might make it a little easier to reorder items as needed, or swap one in for another. Then in `AppTopNav`, iterate over the items and render the `TopNavLink`, using `history` to set the shortcut behavior, setting the appropriate shortcut label/digit based on the index in the array, applying the data-cy className, etc.